### ISSUE TITLE
`removeAsDefaultProtocolClient` does not work on OS X

### ISSUE THREAD

**Issue by bengotow** [electron/electron#5385]
Electron version: `0.37.8`

Using the test code below, I am unable to remove the app as the default mail client. `removeAsDefaultProtocolClient` doesn't seem to do anything - the app remains the default mail client according to Mail.app, and the `isDefaultProtocolClient` function still returns true.

```
require('electron').remote.app.isDefaultProtocolClient('mailto')
> false

require('electron').remote.app.setAsDefaultProtocolClient('mailto')
require('electron').remote.app.isDefaultProtocolClient('mailto')
> true

require('electron').remote.app.removeAsDefaultProtocolClient('mailto')
> false

require('electron').remote.app.isDefaultProtocolClient('mailto')
> true
```

**Comment by bengotow** [comment]
Ahh... well that explains it: https://github.com/electron/electron/blob/d9db23185cba1bc7cedc3b73a56af3093812a726/atom/browser/browser_mac.mm#L44

**Comment by felixrieseberg** [comment]
The trouble is that it seems virtually impossible to actually unregister yourself. I wasn't able to find the opposite of `LSSetDefaultHandlerForURLScheme` - as far as I could figure it out, you have to 

a) uninstall
b) change info.plist
c) set another app as default

**Comment by bengotow** [comment]
Hey! Hmm - we do this in N1 currently by manually opening the plist and
removing the entry entirely, which restores the system default. Maybe the
trick is to call LSSetDefault with null? I'll try it out this weekend - you
may be right...

Ben

**Comment by bengotow** [comment]
Seems like you're right - there's not a great way to reset the default. The behavior when deleting the app seems pretty undefined as well. Mac OS X pick somethings to open mailto:// links, but it's not clear how.

I submitted a PR that implements "removeAsDefaultProtocolClient" to mean "set the default to the first available handler which is not the users app." I  did a bit of testing with mailto, and it seems like `LSCopyAllHandlersForURLScheme` always returns Apple Mail.app first, so I'm hopeful that this will behave as people expect. 😕

**Comment by zcbenz** [comment]
Fixed with #5440.
